lib/puppet/provider/ios_interface/cisco_ios.rb

```diff
@@ -0,0 +1,75 @@
+require_relative '../../util/network_device/cisco_ios/device'
+require_relative '../../../puppet_x/puppetlabs/cisco_ios/utility'
+
+# IOS Interface Puppet Provider for Cisco IOS devices
+class Puppet::Provider::IosInterface::CiscoIos
+  def self.commands_hash
+    @commands_hash ||= PuppetX::CiscoIOS::Utility.load_yaml(File.expand_path(__dir__) + '/command.yaml')
+  end
+
+  def self.instances_from_cli(output)
+    new_instance_fields = []
+    output.scan(%r{#{PuppetX::CiscoIOS::Utility.get_instances(commands_hash)}}).each do |raw_instance_fields|
+      new_instance = PuppetX::CiscoIOS::Utility.parse_resource(raw_instance_fields, commands_hash)
+      new_instance_fields << PuppetX::CiscoIOS::Utility.device_safe_instance(new_instance, commands_hash)
+    end
+    new_instance_fields
+  end
+
+  def self.commands_from_instance(property_hash)
+    property_hash.each do |key, value|
+      property_hash[key] = Puppet::Provider::IosInterface::CiscoIos.false_to_unset(value)
+    end
+    _commands = PuppetX::CiscoIOS::Utility.build_commmands_from_attribute_set_values(property_hash, commands_hash)
+  end
+
+  def self.convert_to_boolean(output)
```

Comment:
This might be better to incorporate in to the utility class, as there is already a `self.convert_no_to_boolean(value)` 

https://github.com/puppetlabs/cisco_ios/blob/master/lib/puppet_x/puppetlabs/cisco_ios/utility.rb#L284

Reply:
Done